src/base/CMakeLists.txt

```diff
@@ -168,3 +168,7 @@ if (APPLE)
     find_library(AppKit_LIBRARY AppKit)
     target_link_libraries(qbt_base PRIVATE ${Carbon_LIBRARY} ${IOKit_LIBRARY} ${AppKit_LIBRARY})
 endif (APPLE)
+
+if (WIN32)
+    target_link_libraries(qbt_base PRIVATE iphlpapi)
```

Comment:
Maybe you should add `link_libraries()` in winconf-msvc.cmake instead?
Or move this code to winconf.cmake if that doesn't work.

Reply:
Iphlapi is specific to qbt_base, but I will move it if you say so

Reply:
>Iphlapi is specific to qbt_base

I guess every lib listed in winconf-mingw.cmake is specific to qbt_base right?
Is it possible to use `target_link_libraries(qbt_base ...)` in both winconf-mingw.cmake and winconf-msvc.cmake?
If not, then I would just use `link_libraries()` in both files for now.

Reply:
`target_link_libraries` statement the will be used will be same for both compilers. So it will be better idea to declare it where qbt_base is defined.

Reply:
>target_link_libraries statement that will be used will be same for both compilers

The lists aren't the same in winconf.pri, there are some differences.

>So it will be better idea to declare it where qbt_base is defined.

OK, but there should be 2 lists like how it is done in winconf.pri.

Reply:
>but there should be 2 lists like how it is done in winconf.pri.

What do you mean?

Reply:
https://github.com/qbittorrent/qBittorrent/blob/master/winconf.pri#L40
https://github.com/qbittorrent/qBittorrent/blob/master/winconf.pri#L48

Reply:
It's due to different library name format. But CMake use only base names which are identical so we don't need to have different lists.

Reply:
Ok done. Tested with msvc (64 bit) and msys2 (64 bit)

Reply:
>It's due to different library name format.

If you look closely, the libraries aren't all identical.

Reply:
>the libraries aren't all identical.

Yes. But it isn't valid reason to have identical things duplicated, IMO. Only really specific things should be conditionally included. Otherwise let's duplicate everything else.
>I wouldn't threat them as the same.

When we talk about Windows system libraries they're the same. The only differences are prefix/suffix but CMake know it.

Reply:
>Only really specific things should be conditionally included. 

There are: crypt32 for msvc only and wsock32, ws2_32 for mingw only.
So this PR should split these special entries.

Reply:
I think CMake not being official BuildSystem, may Chochobo1 is right. I will change accordingly 

Also qt6 will use CMake as official build system and also I think CMake is better then QMake. So you guys should think about CMake more seriously XD>.

Reply:
I think I should go back to `link_libraries` based approach.

Reply:
>some non-sense common/specific lists.

I'm sorry, are we talking about the same thing? The fact that we link the same system libraries regardless of the compiler is nonsense to you?
>Each one is for different compiler and maintained differently

This is a reason for the developer to forget to add something for the "foreign" compiler, as it has been more than once. This PR just corrects such forgetfulness, is not it?

Okay, whatever... no big deal for me until CMake is official qBittorrent build system.

Reply:
>I think I should go back to `link_libraries` based approach.

`link_libraries` is deprecated, AFAIK.

Reply:
> `link_libraries` is deprecated, AFAIK.

Okay, we can go back to target_link_libraries.

> There are: crypt32 required for msvc only and wsock32, ws2_32 required for mingw only.
> So this PR should split these special entries.

I think only require ws2_32 (boost asio) and iphlpapi (ConvertInterfaceNameToLuidW ...)

![image](https://user-images.githubusercontent.com/34717789/68531149-42f93600-0335-11ea-9605-8c34a4c0a837.png)